kontrol-service/engine/flow/merger.go

```diff
@@ -39,3 +39,11 @@ func foldAllIngress(ingresses []*resolved.Ingress) []*resolved.Ingress {
 		return &merged
 	})
 }
+
+func MustGetMarshalledKey[T any](resource T) string {
+	bytes, err := json.Marshal(resource)
```

Comment:
Wondering if we should use the [Marshal](https://pkg.go.dev/k8s.io/api/core/v1#Service.Marshal) functions that are provided by the k8s libs?

Reply:
@lostbean cool I'll try with that

Reply:
Oh! but we are using our custom `Service` object https://github.com/kurtosis-tech/kardinal-kontrol/blob/main/kontrol-service/types/cluster_topology/resolved/core.go#L24